process/transaction/shardProcess.go

```diff
@@ -1216,6 +1231,31 @@ func (txProc *txProcessor) createCompleteEventLog(scr data.TransactionHandler, o
 	}
 }
 
+func (txProc *txProcessor) createRefundSCRForMoveBalance(
+	tx *transaction.Transaction,
+	txHash []byte,
+	originalTxHash []byte,
+	consumedFee *big.Int,
+) error {
+	providedFee := big.NewInt(0).Mul(big.NewInt(0).SetUint64(tx.GasLimit), big.NewInt(0).SetUint64(tx.GasPrice))
+	refundValue := big.NewInt(0).Sub(providedFee, consumedFee)
+
+	refundGasToRelayerSCR := &smartContractResult.SmartContractResult{
+		Nonce:          tx.Nonce,
+		Value:          refundValue,
+		RcvAddr:        tx.RelayerAddr,
+		SndAddr:        tx.SndAddr,
+		PrevTxHash:     txHash,
+		OriginalTxHash: originalTxHash,
+		GasPrice:       tx.GetGasPrice(),
+		CallType:       vm.DirectCall,
+		ReturnMessage:  []byte(core.GasRefundForRelayerMessage),
+		OriginalSender: tx.RelayerAddr,
+	}
+
+	return txProc.scrForwarder.AddIntermediateTransactions([]data.TransactionHandler{refundGasToRelayerSCR})
```

Comment:
err = txProc.createReceiptWithReturnedGas(txHash, tx, acntSrc, moveBalanceCost, totalCost, destShardTxType, isUserTxOfRelayed)
	if err != nil {
		return err
	} 

this is called before. Why would you want to create refundSCRForMoveBalance? A receipt is enough and better.

Reply:
added a new scr because the receipt is not created for user tx of relayed: https://github.com/multiversx/mx-chain-go/blob/75ef2631e72298207500b02251518f3943a32702/process/transaction/shardProcess.go#L356

second reason was to make it compatible with the method used while computing the gasUsed field on api response, that only checks scrs: https://github.com/multiversx/mx-chain-go/blob/75ef2631e72298207500b02251518f3943a32702/outport/process/transactionsfee/transactionChecker.go#L49